Storage/Processor/StreamRange.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace EMS\CommonBundle\Storage\Processor;
+
+use Symfony\Component\HttpFoundation\HeaderBag;
+
+class StreamRange
+{
+    /** @var bool */
+    private $supported;
+    /** @var int */
+    private $fileSize;
+    /** @var int */
+    private $end;
+    /** @var int */
+    private $start;
+
+    public function __construct(HeaderBag $headerBag, int $fileSize)
+    {
+        $this->fileSize = $fileSize;
+        $this->supported = false;
+
+        $range = $headerBag->get('Range');
+        if ($range === null) {
+            return;
```

Comment:
Instead of returning this "incomplete" object. I would throw a custom error here. Then catch it in the Processor.php to simply return the $response.

This would prefent the $supported checks in the other boolean functions.

Reply:
The object is complete RangeHeader are not supported. Adding a throw will implied this refact in the processor class
```
        $streamRange = new StreamRange($request->headers, $fileSize);
        if ($streamRange->isOutOfRange()) {
            return $response;
        }
```
to 
```
        try {
            $streamRange = new StreamRange($request->headers, $fileSize);
        catch (\Exception $e) {
             return $response;
         }
        if ($streamRange->isOutOfRange()) {
            return $response;
        }
```

Reply:
less code?

Reply:
There is less code in the StreamRange class if you do this. The $this->supported property and checks can be removed completely.

I think the refactor is easier to understand. As the StreamRange class becomes cleaner. The exception is also showing that there is a problem with the Range, which is why we return the response.

On creation the IsOutOfRange check could be done by the StreamRange class itself, and also throw an error --> as this would have the same behavior.

IsOutOfRange becomes a private function !

Reply:
I prefer this solution 28cd458a039f019ebf7906c11f757acd0a0d838c